### DISCUSSION TITLE
carregamento de imagem de post no mvc

### DISCUSSION THREAD

**Discussion by jcvilanova** [frontendbr/forum#2281] [Perguntas e Respostas]
Galera, estou tentando fazer um site com PHP e escolhi o modelo tipo Blog, já estou na parte em que os posts sao listados na view conforme a categoria. Aqui me perdi um pouco e fui atras de exemplos na Net, de repente conseui algo que me ajudou, porem, acabei nao entendendo determinada coisa. Quando um artigo é recuperado pelo método ele é identificado pelo ID e este ID é comparado com o numero da pasta onde está a imagem, fiz varios testes e depois de colocar os artigos na ordem correta consegui fazer com que a imagem certa aparecesse no artigo correspondente. Mas como falei, ainda tem algo que está me intrigando. Veja, tenho por exemplo 5 categorias, e a medida que eu for publicando novos artigos cada um deles com sua imagem correspondente, eles estarao no banco de dados em posições variadas, já que posso inserir um artigo de esportes hoje, outro de turismo, outros dois de política, correto? como é autoincremento eles receberão ID automatico e consequentemente serao os ultimos no BD, correto ? Aí vem a pergunta: se a aplicação vai buscar a imagem conforme o ID do artigo e se o artigo que postei por ultimo esta na posição 60, por exemplo, a imagem correspondente vai ter que está na pasta de numero 60, caso contrario a imagem nao sera exibida, certo ? acabei de fazer alguns testes aqui, colocando a imagem chamada espanha.jpg no diretorio 1, quando atualizei a pagina a imagem estava sendo exibida no post de ID 1, levei a imagem para o 3, atualizei a pagina e a imagem passou a ser exibida no post de ID 3. Vejo nisso um problema, não acha ? ou sera se estou entendendo errado ?

porque se meu raciocinio estiver correto eu terei que ter 5000 diretorios, um para cada novo post que eu quiser inserir e isso não faz sentido.

alguem poderia me explicar ?

**Comment by felquis** [comment]
Olá, tudo bem? Deixa eu tentar clarificar algo se possível.


O que entendi da sua explicação:

Cada artigo tem um ID primário único que é incremental,
e as imagens vinculadas a estes artigos serão salvas em pastas nomeadas com o ID do artigo.

Exemplo:

Dado registro de artigos com os IDs 1, 5 e 10 as imagens deste estarão em pastas assim:

```
images/1/imagem.jpg
images/1/outra-imagem.jpg
images/5/foto.jpg
images/5/outra-foto.jpg
images/10/foto.jpg
images/10/imagem.jpg
```

## 1

> Aqui me perdi um pouco e fui atras de exemplos na Net, de repente consegui algo que me ajudou

Compartilha com a gente o link do que te ajudou, pois pode nos ajudar a te ajudar também.

## 2

> fiz varios testes e depois de colocar os artigos na ordem correta consegui
> fazer com que a imagem certa aparecesse no artigo correspondente.

Independentemente de ser um ID sequencial ou não, ou independente da ordem com que os artigos são registrados no banco ou exibidos na tela. O ID do artigo sempre será o mesmo. Portando, você deve sempre pegar o ID do artigo para ir atrás das imagens dele na pasta correspondente.

## 3

> já que posso inserir um artigo de esportes hoje, outro de turismo, outros dois de política,
> correto? como é autoincremento eles receberão ID automatico e consequentemente serao os ultimos no BD,
> correto ?

Sim, o ID pode ser sequencia mas por ter categorias, você verá algo assim acontecer:

```
Artigo ID 1 será da categoria turismo
Artigo ID 2 será da categoria esportes
Artigo ID 3 será da categoria politica
Artigo ID 4 será da categoria turismo
```

## 4

> Aí vem a pergunta: se a aplicação vai buscar a imagem conforme o ID do artigo e se o artigo que postei
> por ultimo esta na posição 60, por exemplo, a imagem correspondente vai ter que está na pasta de numero 60,
> caso contrario a imagem nao sera exibida, certo ?

Sim, faz sentido.


## 5

> acabei de fazer alguns testes aqui,
> colocando a imagem chamada espanha.jpg no diretorio 1,
> quando atualizei a pagina a imagem estava sendo exibida no post de ID 1, levei a imagem para o 3,
> atualizei a pagina e a imagem passou a ser exibida no post de ID 3.
> Vejo nisso um problema, não acha ? ou sera se estou entendendo errado ?

Olha, o resultado do seu teste me faz parecer que o programa está funcionando conforme ele foi criado para funcionar. Então está correto neste contexto.

## 6

> porque se meu raciocinio estiver correto eu terei que ter 5000 diretorios,
> um para cada novo post que eu quiser inserir e isso não faz sentido.

Se você tiver 5 mil diretórios, quer dizer que você tem 5 mil artigos com imagens.
Ter 5 mil diretórios neste caso não é um problema. 

E para ficar claro, você não deve criar 5 mil diretórios manualmente, quando um novo
artigo é adicionado ele o programa deve ser encarregado de criar uma nova pasta usando o ID
do recem criado artigo para armazenar suas imagens.

----

Enfim, espero ter entendido direito e que te ajude.

**Comment by jcvilanova** [comment]
A minha pergunta é: a medida que vou postando mais artigos terei que ter mais pastas com números, correto ?
Não seria meio desautonatizar a coisa ? Imagina que vou postar 10 artigos num dia vou ter que ir lá na estrutura do projeto e criar 10 pastas para receber a imagem desses 10 artigos ?
Isso que não entendi

Enviado do meu iPhone

Em 23 de abr. de 2022, à(s) 13:21, Felquis G ***@***.***> escreveu:

﻿

Olá, tudo bem? Deixa eu tentar clarificar algo se possível.

O que entendi da sua explicação:

Cada artigo tem um ID primário único que é incremental,
e as imagens vinculadas a estes artigos serão salvas em pastas nomeadas com o ID do artigo.

Exemplo:

Dado registro de artigos com os IDs 1, 5 e 10 as imagens deste estarão em pastas assim:

images/1/imagem.jpg
images/1/outra-imagem.jpg
images/5/foto.jpg
images/5/outra-foto.jpg
images/10/foto.jpg
images/10/imagem.jpg


1

Aqui me perdi um pouco e fui atras de exemplos na Net, de repente consegui algo que me ajudou

Compartilha com a gente o link do que te ajudou, pois pode nos ajudar a te ajudar também.

2

fiz varios testes e depois de colocar os artigos na ordem correta consegui
fazer com que a imagem certa aparecesse no artigo correspondente.

Independentemente de ser um ID sequencial ou não, ou independente da ordem com que os artigos são registrados no banco ou exibidos na tela. O ID do artigo sempre será o mesmo. Portando, você deve sempre pegar o ID do artigo para ir atrás das imagens dele na pasta correspondente.

3

já que posso inserir um artigo de esportes hoje, outro de turismo, outros dois de política,
correto? como é autoincremento eles receberão ID automatico e consequentemente serao os ultimos no BD,
correto ?

Sim, o ID pode ser sequencia mas por ter categorias, você verá algo assim acontecer:

Artigo ID 1 será da categoria turismo
Artigo ID 2 será da categoria esportes
Artigo ID 3 será da categoria politica
Artigo ID 4 será da categoria turismo


4

Aí vem a pergunta: se a aplicação vai buscar a imagem conforme o ID do artigo e se o artigo que postei
por ultimo esta na posição 60, por exemplo, a imagem correspondente vai ter que está na pasta de numero 60,
caso contrario a imagem nao sera exibida, certo ?

Sim, faz sentido.

5

acabei de fazer alguns testes aqui,
colocando a imagem chamada espanha.jpg no diretorio 1,
quando atualizei a pagina a imagem estava sendo exibida no post de ID 1, levei a imagem para o 3,
atualizei a pagina e a imagem passou a ser exibida no post de ID 3.
Vejo nisso um problema, não acha ? ou sera se estou entendendo errado ?

Olha, o resultado do seu teste me faz parecer que o programa está funcionando conforme ele foi criado para funcionar. Então está correto neste contexto.

6

porque se meu raciocinio estiver correto eu terei que ter 5000 diretorios,
um para cada novo post que eu quiser inserir e isso não faz sentido.

Se você tiver 5 mil diretórios, quer dizer que você tem 5 mil artigos com imagens.
Ter 5 mil diretórios neste caso não é um problema.

E para ficar claro, você não deve criar 5 mil diretórios manualmente, quando um novo
artigo é adicionado ele o programa deve ser encarregado de criar uma nova pasta usando o ID
do recem criado artigo para armazenar suas imagens.

________________________________

Enfim, espero ter entendido direito e que te ajude.

**Comment by felquis** [comment]
Oi, 

> A minha pergunta é: a medida que vou postando mais artigos terei que ter mais pastas com números, correto ?
> Não seria meio desautonatizar a coisa ? Imagina que vou postar 10 artigos num dia vou ter que ir lá na estrutura do projeto e criar 10 > pastas para receber a imagem desses 10 artigos ?
> Isso que não entendi

Respondi exatamente essa dúvida no item 6 ali em cima, da uma olhada.

**Comment by jcvilanova** [comment]
Pois pronto, agora vem a pergunta, e como se faz essa automatização que relaciona artigo criado a uma nova pasta? Quer dizer que só criar um novo artigo o sistema vai criar uma nova pasta automaticamente ?

Enviado do meu iPhone

Em 23 de abr. de 2022, à(s) 19:12, Felquis G ***@***.***> escreveu:

﻿

Oi,

A minha pergunta é: a medida que vou postando mais artigos terei que ter mais pastas com números, correto ?
Não seria meio desautonatizar a coisa ? Imagina que vou postar 10 artigos num dia vou ter que ir lá na estrutura do projeto e criar 10 > pastas para receber a imagem desses 10 artigos ?
Isso que não entendi

Respondi exatamente essa dúvida no item 6 ali em cima, da uma olhada.

**Comment by felquis** [comment]
> como se faz essa automatização que relaciona artigo criado a uma nova pasta

O tutorial que você está seguindo não faz isso? Manda o link pra gente.

**Comment by jcvilanova** [comment]
Não. Não faz. Na verdade as os vídeos estão incompletos. Tem aula 6,7,8 por exemplo aí salta para a 11. Lá na frente faltam duas três aulas de novo. Aí assim vai
Tenho penado para resolver. Com ajuda de vocês em alguns fóruns eu tenho feito. Graças a Deus, mas tem coisa que aperta.
Por exemplo essa aí de automatizar a criação das pastas eu nem imagino como se faz

Enviado do meu iPhone

Em 24 de abr. de 2022, à(s) 14:27, Felquis G ***@***.***> escreveu:

﻿

como se faz essa automatização que relaciona artigo criado a uma nova pasta

O tutorial que você está seguindo não faz isso? Manda o link pra gente.
